compiler/rustc_session/src/session.rs

```diff
@@ -1540,6 +1540,14 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
             break;
         }
     }
+
+    if sess.opts.debugging_opts.mir_opt_stats {
+        if let Err(_) = rustc_data_structures::statistics::try_enable() {
+            sess.warn(&format!(
+                "-Zmir-opt-stats is only available when rustc is compiled with debug-assertions"
+            ));
```

Comment:
This should be an error I instead think.

Reply:
Replaced with an error.